lib/build.js

```diff
@@ -1,208 +1,133 @@
-const JobModel = require('./job');
-const UserModel = require('./user');
+const JobFactory = require('./jobFactory');
+const UserFactory = require('./userFactory');
 const githubHelper = require('./github');
 
-/**
- * Given a Pipeline ID, look up the associated SCM Url
- * @method fetchScmUrl
- * @param  {Model }    pipeline           Instance of the Pipeline model
- * @param  {Object}    job                The job object
- * @param  {String}    job.pipelineId     The ID of the pipeline to look at
- * @param  {String}    params.name        The name of the job we are building
- * @param  {Function}  callback           fn(err, result), where result is an object consisting of two
- *                                        keys: "pipelineID", "scmUrl", and "jobName"
- */
-function fetchScmUrl(pipeline, job, callback) {
-    const pipelineId = job.pipelineId;
-
-    pipeline.get(pipelineId, (err, data) => {
-        if (err) {
-            return callback(err);
-        }
-
-        return callback(null, {
-            pipelineId,
-            jobName: job.name,
-            scmUrl: data.scmUrl
-        });
-    });
-}
-
-/**
- * Look up repo info such as user, repo name, and sha
- * @method getRepoInfo
- * @param  {Model}    user          The user model
- * @param  {Object}   config        Configuration object that might/might not include sha
- * @param  {Object}   buildInfo     Other build info that will be returned. Includes
- *                                  pipelineId, jobName, and scmUrl
- * @param  {Function} callback      fn(err, sha)
- */
-function getRepoInfo(user, config, buildInfo, callback) {
-    const repoInfo = githubHelper.getInfo(buildInfo.scmUrl);
-    const response = hoek.applyToDefaults(buildInfo, repoInfo);
-
-    if (config.sha) {
-        return callback(null, hoek.applyToDefaults(response, { sha: config.sha }));
-    }
-
-    return githubHelper.run({
-        user,
-        username: config.username,
-        action: 'getBranch',
-        params: repoInfo
-    }, (err, repo) => callback(err, hoek.applyToDefaults(response, { sha: repo.commit.sha })));
-}
-
 class BuildModel extends BaseModel {
     /**
      * Construct a BuildModel object
      * @method constructor
-     * @param  {Object}    datastore         Object that will perform operations on the datastore
-     * @param  {Object}    executor          Object that will perform executor operations
-     * @param  {String}    password          Login password
+     * @param  {Object}    config
+     * @param  {Object}    config.datastore         Object that will perform operations on the datastore
+     * @param  {Object}    config.executor          Object that will perform executor operations
+     * @param  {String}    config.password          Login password
+     * @param  {String}    config.username          The user that created this build
+     * @param  {String}    config.jobId             The ID of the associated job to start
```

Comment:
The model's constructors should take in all the data that is contained in the row of a table for that resource. The data-schema defines the list of all the keys that are in a row of a table.